add Minus operator

m-schen opened the pull request:
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #4094 

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

fengttt commented on pkg/sql/colexec/minus/minus.go:
```diff
@@ -0,0 +1,193 @@
+		}
+	}
+	{
+		// aligning the decimal precision
```
Decimal does not need to be aligned any more.

fengttt reviewed (changes requested):
Can please you comment in minus.go, describe what you do.   How do you do bookkeeping in the hashtable, etc.

This kind a change should have a bvt test case.

m-schen commented:
> Can please you comment in minus.go, describe what you do. How do you do bookkeeping in the hashtable, etc.
> 
> This kind a change should have a bvt test case.

I have described how I did it in the comments of the call function.  
i will make a doc to describe how to use the hashtable next week.
this pr can not have bvt test because it is just an operator implement but without plan build. it is hard to do test end to end.

nnsgmsone commented on pkg/sql/colexec/minus/minus.go:
```diff
@@ -0,0 +1,193 @@
+			if n > hashmap.UnitLimit {
+				n = hashmap.UnitLimit
+			}
+			vs, _ := itr.Insert(i, n, bat.Vecs, ctr.scales[index])
```
The likelihood of doing so would be very low and I would advise against it.

m-schen commented:
I've added comments to the minus.go file to describe how I do the minus work and removed the useless scalar related work. And bvt test can't be done at now because it is just an operator implement but without plan built.

I will push a hash table related doc today but not this pull request.

m-schen commented:
I will make another pull request first, to reduce the cost of learning to use hash tables.
and change related codes about the use of hash table at this pr.

m-schen commented:
> I want to make another pull request first, to reduce the cost of learning to use hash tables.
> and change related codes about the use of hash table at this pr.
